src/lib/code-sample-editor.ts

```diff
@@ -434,3 +485,22 @@ const mimeTypeToTypeEnum = (mimeType?: string) => {
   }
   return undefined;
 };
+
+const typeEnumToMimeType = (type?: string) => {
+  // TODO: infer type based on extension too
+  if (type === undefined) {
+    return;
+  }
+  switch (type) {
+    // TypeScript
+    case 'ts':
+      return 'video/mp2t';
```

Comment:
Video/mp2t doesn't seem right?

Reply:
See the comment above. It's the registered mime-type for .ts files and what I'm seeing returned by servers. afaict there's no mime-type for TypeScript.